foca/utils/logging.py

```diff
@@ -0,0 +1,78 @@
+"""Decorator for logging requests and responses."""
+
+import logging
+from connexion import request
+from functools import wraps
+from typing import (Callable, Optional, )
+
+logger = logging.getLogger(__name__)
+
+
+def log_traffic(_fn: Callable = None,
+                log_request: Optional[bool] = True,
```

Comment:
I think you may have misunderstood the meaning of `Optional[]`. It doesn't have anything to do whether an argument is required - it says that the _type_ of the argument can be either `None` or the type indicated in between the angular brackets.

In fact, the only argument that has an `Optional[]` type is `_fn` (which can be a callable or `None`) - whereas all the others are _always_ either `bool` (`log_request` and `log_response`) or `int` (`log_level`).

To be more explicit, I have changed the `log_level` default from `20` to `logging.INFO`.

Reply:
I definitely had misunderstood it; thanks for the clarification